var/spack/repos/builtin/packages/selalib/package.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Selalib(CMakePackage):
+    """SeLaLib is a modular library for the kinetic and gyrokinetic simulation
+    of tokamak plasmas by the semi-lagrangian or particle-in-cell methods"""
+
+    homepage = "https://selalib.github.io/selalib"
+    url = "https://github.com/selalib/selalib"
+    git = "https://github.com/selalib/selalib"
+
+    maintainers("pnavaro", "freifrauvonbleifrei")
+
+    version("main", branch="main")
+
+    variant("fmempool", default=False, description="Use memory pool")
+    variant("mpi", default=True, description="Build with MPI support")
+    variant("openmp", default=True, description="Build with OpenMP support")
+    variant("compression", default=False, description="Add compression by ZFP")
+
+    requires(
+        "%gcc@9.0.0:",
+        "%clang@16.0.0:",
+        "%intel@18.0:",
+        "%oneapi@18.0:",
+        policy="one_of",
+        msg="SeLaLib requires new-enough Fortran compiler",
+    )
+
+    depends_on("cmake@3.6.0:", type=("build"))
+    depends_on("blas")
+    depends_on("fftw")
+    depends_on("fftw+mpi", when="+mpi")
+    depends_on("fftw+openmp", when="+openmp")
+    depends_on("fgsl")
+    depends_on("git", type=("build", "run", "test"))
+    depends_on("hdf5+fortran+cxx")
```

Comment:
Do we need:
```python
depends_on("hdf5+mpi", when="+mpi")
```
? You can also consider grouping all the requirements for `mpi` under:
```python
with when("mpi"):
    depends_on("mpi")
    depends_on("hdf5+mpi")
    depends_on("fftw+mpi")
    ...
```

Reply:
yes, I like the grouping idea, thx!

Reply:
turns out that the grouped syntax led to the concretizer not considering mpi at all :/ I have reverted to inline `when("mpi")`.

Reply:
Typo on my side:
```
with when("+mpi"):
    ...
```